src/ez/helper/ForeignKeyConstraint.java

```diff
@@ -0,0 +1,60 @@
+package ez.helper;
+
+import com.google.common.base.Strings;
+
+import ez.DB;
+import ez.misc.DatabaseType;
+
+import ox.Log;
+
+public class ForeignKeyConstraint {
+
+  protected String foreigKeyName, sourceTable, sourceColumnName, foreignTable, foreignColumnName;
+
+  protected ForeignKeyConstraint(String sourceTable, String sourceColumnName, String foreignTable,
+      String foreignColumnName) {
+    this.sourceTable = sourceTable;
+    this.sourceColumnName = sourceColumnName;
+    this.foreignTable = foreignTable;
+    this.foreignColumnName = foreignColumnName;
+  }
+
+  protected ForeignKeyConstraint(String sourceTable, String sourceColumnName, String foreignTable,
+      String foreignColumnName, String foreignKeyName) {
+    this.sourceTable = sourceTable;
```

Comment:
this should call normalize() on all arguments.  It should also checkNotEmpty() on once that it expects to have values

Reply:
done